Library/ViewModels/LoginToutViewModel.swift

```diff
@@ -1,12 +1,29 @@
+import AuthenticationServices
 import FBSDKLoginKit
 import KsApi
 import Prelude
 import ReactiveSwift
 
+public typealias SignInWithAppleData = (appId: String, firstName: String?, lastName: String?, token: String)
+
 public protocol LoginToutViewModelInputs {
+  /// Call when Continue withApple button is pressed
+  func appleLoginButtonPressed()
+
   /// Call to set the reason the user is attempting to log in
   func configureWith(_ intent: LoginIntent, project: Project?, reward: Reward?)
 
+  /// Call when Apple completes authorization
+  @available(iOS 13.0, *)
+  func appleAuthorizationDidComplete(with data: SignInWithAppleData?)
+
+  /// Call when Apple completes authorization with error
+  @available(iOS 13.0, *)
+  func appleAuthorizationDidComplete(with error: Error)
```

Comment:
Maybe a better name for this would be `appleAuthorizationDidFail(with error: Error)`? That way you can differentiate between a success and a failure. Above, you could also renamed `appleAuthorizationDidComplete` to `appleAuthorizationDidSucceed`.

Reply:
Agreed!